src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -59,136 +61,91 @@ const defaultProps = {
+            tooltipHeight,
+            tooltipContentWidth,
+            props.shiftHorizontal,
+            props.shiftVertical,
+        ),
+        [props.animation, props.windowWidth, props.xOffset, props.yOffset, props.wrapperWidth, props.wrapperHeight,
+            props.maxWidth, tooltipWidth, tooltipHeight, tooltipContentWidth, props.shiftHorizontal, props.shiftVertical],
+    );
+
+    let content;
+    if (props.renderTooltipContent) {
+        content = (
+            <View ref={contentRef}>
+                {props.renderTooltipContent()}
+            </View>
         );
-
-        const contentRef = (ref) => {
-            // Once the content for the tooltip first renders, update the width of the tooltip dynamically to fit the width of the content.
-            // Note that we can't have this code in componentDidMount because the ref for the content won't be set until after the first render
-            if (this.contentRef) {
-                return;
-            }
-
-            this.contentRef = ref;
-            this.updateTooltipContentWidth();
-        };
-
-        let content;
-        if (this.props.renderTooltipContent) {
-            content = (
-                <View ref={contentRef}>
-                    {this.props.renderTooltipContent()}
-                </View>
-            );
-        } else {
-            content = (
-                <Text numberOfLines={this.props.numberOfLines} style={tooltipTextStyle}>
-                    <Text style={tooltipTextStyle} ref={contentRef}>
-                        {this.props.text}
-                    </Text>
+    } else {
+        content = (
+            <Text numberOfLines={props.numberOfLines} style={tooltipTextStyle}>
+                <Text style={tooltipTextStyle} ref={contentRef}>
+                    {props.text}
                 </Text>
-            );
-        }
-
-        return ReactDOM.createPortal(
-            <Animated.View
-                onLayout={this.measureTooltip}
-                style={[tooltipWrapperStyle, animationStyle]}
-            >
-                {content}
-                <View style={pointerWrapperStyle}>
-                    <View style={pointerStyle} />
-                </View>
-            </Animated.View>,
-            document.querySelector('body'),
+            </Text>
         );
     }
-}
+
+    return ReactDOM.createPortal(
+        <Animated.View
+            ref={wrapper}
+            style={[tooltipWrapperStyle, animationStyle]}
+        >
+            {content}
+            <View style={pointerWrapperStyle}>
+                <View style={pointerStyle} />
+            </View>
+        </Animated.View>,
+        document.querySelector('body'),
+    );
+};
 
 TooltipRenderedOnPageBody.propTypes = propTypes;
 TooltipRenderedOnPageBody.defaultProps = defaultProps;
+TooltipRenderedOnPageBody.displayName = 'TooltipRenderedOnPageBody';
 
-export default TooltipRenderedOnPageBody;
+export default React.memo(TooltipRenderedOnPageBody);
```

Comment:
NAB, but you've done a great job of explaining the _why_ of what's going on in https://github.com/Expensify/App/pull/18189#issuecomment-1528761260, it would be nice to get a condensed version in here. Maybe something like:

```
We pass a key of `this.props.text`, so whenever the text changes this component will completely remount with a fresh state. This prevents flickering/moving while remaining performant.
```

Does that accurately summarize? If not, feel free to rewrite.

Reply:
The comment sounds good, but I think you mean to put it above the `key` props. I also adjusted it with the fact that we also pass `renderTooltipContentKey`.